.github/workflows/ci.yml

```diff
@@ -72,25 +72,94 @@ jobs:
 
       - name: Build debug builds
         shell: bash
-        run: cargo build
+        run: cargo build --workspace
 
       - name: Upload criticalup debug build
         uses: actions/upload-artifact@v4
         with:
           name: criticalup-${{ matrix.os }}
-          path: target/debug/criticalup
+          path: target/debug/criticalup${{ (runner.os == 'Windows' && '.exe') || '' }}
+          retention-days: 7
 
       - name: Upload upload criticalup-dev debug build
         uses: actions/upload-artifact@v4
         with:
           name: criticalup-dev-${{ matrix.os }}
-          path: target/debug/criticalup-dev
+          path: target/debug/criticalup-dev${{ (runner.os == 'Windows' && '.exe') || '' }}
+          retention-days: 7
 
       - name: Upload cargo timings
         uses: actions/upload-artifact@v4
         with:
           name: cargo-timings-${{ matrix.os }}
           path: target/cargo-timings/cargo-timing.html
+          retention-days: 7
+
+  runner-test:
+    name: Test CriticalUp on GHA Runners
+    needs: [build-test]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      CRITICALUP_TOKEN: ${{ secrets.CRITICALUP_TOKEN }}
+    steps:
+      - name: "Define toolchain"
+        shell: bash # Powershell does not do heredocs
+        run: |
+          mkdir -p crab-boil
+          cd crab-boil
+          cat <<- EOF > criticalup.toml
+            manifest-version = 1
+
+            [products.ferrocene]
+            release = "stable-24.11.0"
+            packages = [
+                "cargo-\${rustc-host}",
+                "rustc-\${rustc-host}",
+                "rust-std-\${rustc-host}",
+            ]
+          EOF
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: criticalup-${{ matrix.os }}
+          path: crab-boil
+
+      - name: Make executable
+        if: ${{ runner.os != 'Windows' }}
+        working-directory: crab-boil
+        run: chmod +x ./criticalup
+
+      - name: Auth Criticalup
+        working-directory: crab-boil
+        run: ./criticalup auth set $CRITICALUP_TOKEN
+
+
+      - name: Install toolchain
+        working-directory: crab-boil
+        run: |
+          ./criticalup install
+
+      - name: Run test workflow
+        working-directory: crab-boil
+        run: |
+          ./criticalup run -- cargo init
+
+          ${{ (runner.os == 'Windows' && './criticalup.exe run -- cargo --version') || '' }}
+          ${{ (runner.os == 'Windows' && './criticalup run -- cargo.exe --version') || '' }}
```

Comment:
Can we run this as a separate "windows" step?

Reply:
Great idea.